shared/src/price_estimate.rs

```diff
@@ -184,6 +183,22 @@ impl PriceEstimating for BaselinePriceEstimator {
     }
 }
 
+fn amounts_to_price(
+    sell_amount: U256,
+    buy_amount: U256,
+) -> Result<BigRational, PriceEstimationError> {
+    // Due to integer rounding it is possible that we find path but the amount is 0.
+    if sell_amount.is_zero() || buy_amount.is_zero() {
```

Comment:
just curious, what's wrong with allowing sell_amount=0 ?

Reply:
I guess  technically we can allow it but it doesn't make sense to me that you would have a 0 price. Currently the price is always sell / buy but we could have arbitrarily chosen buy / sell too in which case it would be division by 0 again.